File: Advanced Programming Methods/Toy Language/src/Model/DataStructures/MyList.java
package Model.DataStructures;

import java.util.ArrayDeque;
import java.util.Queue;

//Out Table implemented as an array deque because we needed the most recent output on the first positions
public class MyList<T> implements IList<T> {
    private Queue<T> list;

    public MyList(){
        this.list = new ArrayDeque<>();
    }

    @Override
    public void add(T el) {
        list.add(el);
    }

    @Override
    public T get()throws MyException {
        if(list.isEmpty())
            throw new MyException("Empty Out Array!");
        return this.list.peek();
    }

    @Override
    public int size() {
        return list.size();
    }

    @Override
    public String toString(){
        String res = "Out:";
        for(T el:list){
            res = res.concat(System.lineSeparator());
            res = res.concat(el.toString());
        }
        return res;
    }
}

File: Advanced Programming Methods/Toy Language C#/A2/Model/Data Structures/MyList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Model.Data_Structures
{
    //Out Table implemented as an array deque because we needed the most recent output on the first positions
    public class MyList<T> : IList<T> {
        private Queue<T> list;

        public MyList()
        {
            list = new Queue<T>();
        }
    
        public void add(T el)
        {
            list.Enqueue(el);
        }
    
        public T get()
        {
            if(list.Count() == 0)
                throw new MyException("Empty Out Array!");
            return list.Peek();
        }
    
        public int size()
        {
            return list.Count();
        }
    
        public override String ToString()
        {
            String res = "Out:";
            foreach (T el in list)
            {
                res += Environment.NewLine;
                res += el.ToString();
            }
            return res;
        }
    }

}
